internal/cli/cmd/demo/demo.go

```diff
@@ -40,20 +40,27 @@ func NewCmd() *cobra.Command {
 	var clusterType string
 	var wait bool
 	var err error
+	var prompt any
 	cmd := &cobra.Command{
 		Use:   "demo <subcommand>",
 		Short: "deploy demo ob cluster and tenant in easier way",
 		Long:  `deploy demo ob cluster and tenant in easier way, currently support single node and three node cluster, with corresponding tenant`,
 		Args:  cobra.NoArgs,
 		PreRun: func(cmd *cobra.Command, args []string) {
 			// prompt for cluster create options
-			prompt := pf.CreatePrompt(cluster.FLAG_NAME)
-			if clusterOptions.Name, err = pf.RunPromptE(prompt); err != nil {
-				logger.Fatalln(err)
-			}
-			prompt = pf.CreatePrompt(cluster.FLAG_NAMESPACE)
-			if clusterOptions.Namespace, err = pf.RunPromptE(prompt); err != nil {
-				logger.Fatalln(err)
+			for {
+				prompt = pf.CreatePrompt(cluster.FLAG_NAME)
+				if clusterOptions.Name, err = pf.RunPromptE(prompt); err != nil {
+					logger.Fatalln(err)
+				}
+				prompt = pf.CreatePrompt(cluster.FLAG_NAMESPACE)
+				if clusterOptions.Namespace, err = pf.RunPromptE(prompt); err != nil {
+					logger.Fatalln(err)
+				}
+				if !utils.CheckIfClusterExists(cmd.Context(), clusterOptions.Name, clusterOptions.Namespace) {
+					break
+				}
+				logger.Printf("Cluster %s already exists in namespace %s, please input another cluster name", clusterOptions.Name, clusterOptions.Namespace)
```

Comment:
This `for-loop` seems to be endless if the flow gets here.

Reply:
Yes, because here we want to check if the cluster name typed by user is available in the corresponding namespace, and if it is not exists (which is meet the requirements, the return value of `utils.CheckIfClusterExists` is `false` ), the workflow will be advanced.

Reply:
OK, I see. The cli tool is asking the user to enter the name and namespace of the cluster here.